storage/innobase/xtrabackup/src/ds_decrypt.c

```diff
@@ -358,19 +358,23 @@ decrypt_write(ds_file_t *file, const void *buf, size_t len)
 		thd = threads;
 
 		pthread_mutex_lock(&thd->ctrl_mutex);
+		parse_result = thd->parse_result;
 
 		do {
+			size_t new_data_size = 0;
 			if (parse_result == XB_CRYPT_READ_INCOMPLETE) {
-				crypt_file->buf_size = crypt_file->buf_size * 2;
+				crypt_file->buf_size = crypt_file->buf_size + len;
```

Comment:
I didn't want to copy whole input buffer here, I wanted to copy just a some data from the beginning to complete the chunk.

Reply:
fixed